cpp/src/arrow/util/compression.h

```diff
@@ -107,6 +107,44 @@ class ARROW_EXPORT Decompressor {
   // XXX add methods for buffer size heuristics?
 };
 
+/// \brief Compression codec options
+class ARROW_EXPORT CodecOptions {
+ public:
+  explicit CodecOptions(int compression_level = kUseDefaultCompressionLevel) {
+    compression_level_ = compression_level;
+  }
+  virtual ~CodecOptions() = default;
+
+  int compression_level_;
```

Comment:
So would this be changed to a `std::nullopt`?

Reply:
yeah I made some changes to these part, will push those soon.